articles/2023-11-17-vanilla-os-orchid-devlog.md

```diff
@@ -0,0 +1,80 @@
+
+Flags:
+  -h, --help   help for android
+
+Use "vso android [command] --help" for more information about a command.
+```
+
+## F-Droid
+
+[F-Droid](https://f-droid.org/) is a popular repository for installing Free and Open Source Android applications.
+
+The Waydroid container is integrated with F-Droid to provide a seamless experience. The container first reads any listed repositories (by default we include F-Droid and [IzzyOnDroid](https://apt.izzysoft.de/fdroid/)) and syncs the index-v2 files if they are outdated or not already downloaded.
+
+The index-v2 files are JSON files containing information about every single app available in the repository it was downloaded from, allowing VSO to search and display information about any app without having to frequently send requests to an API.
+
+One main consideration with this approach is the size of the repository; the indexes can become quite large (F-Droid index-v2 for example is approximately 36MB), that's why VSO caches the entire index locally and automatically refreshes it once every week.
+
+When a user searches or tries to install an app with these indexes saved, all VSO has to do is parse the index-v2 files and search them for the name of the application. The user can define both the reverse domain name notation (e.g. org.fdroid.fdroid) or the name displayed to the user (just "fdroid"), VSO will automatically match for both names regardless of which version has been passed by the user to provide the highest chance of finding an app.
+
+If only one match is found, VSO does not prompt the user and proceeds with the found app, if multiple matches are found, VSO asks the user which app they want VSO process with a multiple choice question. The question will show the normal name, the rdns name, the summary of the app, and in which repository it was found to make the apps more distinguishable for the user.
+
+Once the user selects the app to process, depending on what operation the user executed different things are done:
+
+- remove - the app gets deleted with the `ewaydroid app remove <app rdns>` command.
+
+- info - the information of the app gets printed out
+
+- install - VSO uses the rdns name to make an API call to `https://<repourl>/api/v1/packages/<rdns>`, this returns a JSON file containing some app information, one of those being the suggested version code, based on that VSO builds the URL to download the Apk and stores it in the cache. Once the Apk is downloaded, it is simply installed using `ewaydroid app install ~/.cache/vso/apks/<rdns>_<versioncode>.apk`. In addition to that, VSO also maintains a [bbolt](https://pkg.go.dev/go.etcd.io/bbolt) database which contains information about every app the user has currently installed. This allows for a quicker removal of apps, as we can just check the database for installed apps instead of having to use Waydroid to get the list of apps, convert it to a readable format, and filter out system apps.
```

Comment:
```suggestion
- install - VSO uses the rdns name to make an API call to `https://<repourl>/api/v1/packages/<rdns>`, this returns a JSON file containing some app information, one of those being the suggested version code. Based on that, VSO builds the URL to download the Apk and stores it in the cache. Once the Apk is downloaded, it is simply installed using `ewaydroid app install ~/.cache/vso/apks/<rdns>_<versioncode>.apk`. In addition to that, VSO also maintains a [bbolt](https://pkg.go.dev/go.etcd.io/bbolt) database which contains information about every app the user has currently installed. This allows for a quicker removal of apps, as we can just check the database for installed apps instead of having to use Waydroid to get the list of apps, convert it to a readable format, and filter out system apps.
```